{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c91b32f1-e44f-4892-bcbd-515670db9b51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Go\n"
     ]
    }
   ],
   "source": [
    "print(\"Go\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "52565a04-aa25-4f3b-99f4-7c1f55077fef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Channels:\n",
      " - defaults\n",
      " - conda-forge\n",
      "Platform: osx-arm64\n",
      "doneecting package metadata (repodata.json): - \n",
      "doneing environment: - \n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/anaconda3/envs/Recipe_Chatbot\n",
      "\n",
      "  added / updated specs:\n",
      "    - pymongo\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages:\n",
      "\n",
      "donearing transaction: - \n",
      "donefying transaction: | \n",
      "doneuting transaction: \\ \n"
     ]
    }
   ],
   "source": [
    "!conda install --force-reinstall pymongo -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "21c50d07-adc4-4ef2-8f6a-321ff5f0fd6a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pymongo==3.12 (from pymongo[srv]==3.12)\n",
      "  Using cached pymongo-3.12.0-cp313-cp313-macosx_11_0_arm64.whl\n",
      "Requirement already satisfied: dnspython<2.0.0,>=1.16.0 in /opt/anaconda3/envs/Recipe_Chatbot/lib/python3.13/site-packages (from pymongo[srv]==3.12) (1.16.0)\n",
      "Installing collected packages: pymongo\n",
      "  Attempting uninstall: pymongo\n",
      "    Found existing installation: pymongo 4.11\n",
      "    Uninstalling pymongo-4.11:\n",
      "      Successfully uninstalled pymongo-4.11\n",
      "Successfully installed pymongo-3.12.0\n"
     ]
    }
   ],
   "source": [
    "!python -m pip install \"pymongo[srv]\"==3.12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "eed8d9b1-0014-4787-83ad-5a2171d0a863",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster0-shard-00-01.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018),cluster0-shard-00-00.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018),cluster0-shard-00-02.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018), Timeout: 30s, Topology Description: <TopologyDescription id: 679a5072b1c917469e1c63a7, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('cluster0-shard-00-00.riovx.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('cluster0-shard-00-00.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018)')>, <ServerDescription ('cluster0-shard-00-01.riovx.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('cluster0-shard-00-01.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018)')>, <ServerDescription ('cluster0-shard-00-02.riovx.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('cluster0-shard-00-02.riovx.mongodb.net:27017: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1018)')>]>\n"
     ]
    }
   ],
   "source": [
    "from pymongo.mongo_client import MongoClient\n",
    "from pymongo.server_api import ServerApi\n",
    "import urllib.parse\n",
    "\n",
    "username = urllib.parse.quote_plus(\"MayankMadan\")\n",
    "password = urllib.parse.quote_plus(\"dgAbr@k@d@br@snc2004\")\n",
    "\n",
    "uri = uri = f\"mongodb+srv://{username}:{password}@cluster0.riovx.mongodb.net/?authSource=admin&retryWrites=true&w=majority&appName=Cluster0\"\n",
    "\n",
    "# Create a new client and connect to the server\n",
    "client = MongoClient(uri,server_api=ServerApi('1'))\n",
    "\n",
    "# Send a ping to confirm a successful connection\n",
    "try:\n",
    "    client.admin.command('ping')\n",
    "    print(\"Pinged your deployment. You successfully connected to MongoDB!\")\n",
    "except Exception as e:\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "367b9736-670e-4012-be3c-7294e2ac8d28",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "103.98.191.88"
     ]
    }
   ],
   "source": [
    "!curl ifconfig.me"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8df9f1dd-17a0-4f6e-9a2e-49ced0be6ba9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
